compiler/damlc/daml-ide-core/test/Development/IDE/Core/API/Testing/Visualize.hs

```diff
@@ -0,0 +1,65 @@
+-- Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Development.IDE.Core.API.Testing.Visualize
+    ( ExpectedGraph(..)
+    , ExpectedSubGraph(..)
+    , ExpectedChoiceDetails(..)
+    , FailedGraphExpectation(..)
+    , graphTest
+    )
+    where
+
+import Control.Monad
+import Data.Bifunctor
+import qualified Data.NameMap as NM
+import qualified Data.Text as T
+
+import qualified DA.Daml.LF.Ast as LF
+import qualified DA.Daml.Visual as V
+
+type TemplateName = String
+type ChoiceName = String
+
+data ExpectedGraph = ExpectedGraph
+    { expectedSubgraphs :: [ExpectedSubGraph]
+    , expectedEdges :: [(ExpectedChoiceDetails, ExpectedChoiceDetails)]
+    } deriving (Eq, Ord, Show )
+
+data ExpectedSubGraph = ExpectedSubGraph
+    { expectedNodes :: [ChoiceName]
+    , expectedTplFields :: [String]
+    , expectedTemplate :: TemplateName
+    } deriving (Eq, Ord, Show )
+
+data ExpectedChoiceDetails = ExpectedChoiceDetails
```

Comment:
I know that the code here is rather confusing (why is there no template? :confused:) but I deliberately tried to not touch more than absolutely necessary.